src/stuff.m4

```diff
@@ -1243,7 +1243,10 @@ m4_define([_MAKE_DEFAULTS_OPTIONAL], [m4_do(
 			[action], [],
 			[incr], [_arg_varname=m4_expand(_default)_ENDL_],
 			[repeated], [_arg_varname=(_default)_ENDL_],
-			[_arg_varname=_sh_quote(_default)_ENDL_])],
+			[m4_do(
+				[_arg_varname=_sh_quote(_default)_ENDL_],
+				[m4_list_contains([HAVE_SUPPLIED], _argname, [[_supplied]_arg_varname=0]_ENDL_, [])],
+			)])],
```

Comment:
This is inside a `case` statement, which is probably not what you intended. What you want is to perform this check every time, even for `incr` or `repeated` arguments. So you can basically put `[m4_list_contains([HAVE_SUPPLIED], _argname, [[_supplied]_arg_varname=0]_ENDL_, [])],` after the `m4_case` to the same level of indentation as `m4_case`.

Reply:
I was trying to come up with a use-case for INCR/REPEATED and don't see a solid use-case. Any ideas? I can add it anyway since we do declare all OPTIONALs are supported.

Also, putting it after m4_case will allow positional arguments to also have a `_supplied_arg_...`, I think we should fail in this case because a _supplied_arg for a non-optional doesn't make sense.